compiler/x/codegen/X86Ops.hpp

```diff
@@ -188,6 +188,15 @@ inline TR_X86OpCodes SubMemReg  (bool is64Bit, bool isWithBorrow) { return isWit
 inline TR_X86OpCodes SubRegReg  (bool is64Bit, bool isWithBorrow) { return isWithBorrow  ? SBBRegReg(is64Bit)  : SUBRegReg(is64Bit)  ; }
 inline TR_X86OpCodes SubRegMem  (bool is64Bit, bool isWithBorrow) { return isWithBorrow  ? SBBRegMem(is64Bit)  : SUBRegMem(is64Bit)  ; }
 
+// Data type based opcodes
+inline TR_X86OpCodes IMulRegReg (bool is64Bit, bool isShort) { return isShort ? IMUL2RegReg : IMULRegReg(is64Bit); }
+inline TR_X86OpCodes IMulRegMem (bool is64Bit, bool isShort) { return isShort ? IMUL2RegMem : IMULRegMem(is64Bit); }
+inline TR_X86OpCodes MovRegReg  (bool is64Bit, bool isShort) { return isShort ? MOV2RegReg : MOVRegReg(is64Bit); }
+inline TR_X86OpCodes IMulRegRegImms (bool is64Bit, bool isShort) { return isShort ? IMUL2RegRegImms : IMULRegRegImms(is64Bit); }
+inline TR_X86OpCodes IMulRegRegImm4 (bool is64Bit, bool isShort) { return isShort ? IMUL2RegRegImm2 : IMULRegRegImm4(is64Bit); }
+inline TR_X86OpCodes IMulRegMemImms (bool is64Bit, bool isShort) { return isShort ? IMUL2RegMemImms : IMULRegMemImms(is64Bit); }
+inline TR_X86OpCodes IMulRegMemImm4 (bool is64Bit, bool isShort) { return isShort ? IMUL2RegMemImm2 : IMULRegMemImm4(is64Bit); }
+
```

Comment:
We need to give more thoughts to those APIs. Asking the user to pass 2 booleans both related to the data type is confusing.  The `is64Bit` boolean exists for historical reasons that we used to only need to consider 32bit and 64bit form of the x86 opcodes. Now we need to take more data sizes into consideration and need to redesign the APIs here to get the opcode which is out of the scope of this item. https://github.com/eclipse/omr/blob/367db1e6e713de08b0a3604783a0ce4b445f68a9/compiler/x/codegen/X86Ops.hpp#L43-L53 

For now, I think we can just pass  `node->getSize()` as the only argument to the newly added `inline TR_X86OpCodes xxx` APIs to make them something like `inline TR_X86OpCodes MovRegReg  (int size)` and the caller can use the API by `TR_X86OpCodes  op = MovRegReg(node->getSize())`
@andrewcraik  Hey Andrew, do you have any suggestions for this one?

Reply:
I agree with @cathyzhyi that we should not have two bools - quite ugly. Some possible choices for the input parameter:

- the size in bytes
- the node type (address, int64, int32 etc)
- the node

I think the node is too much information and too high-level for this kind of API. The size in bytes is probably the right thing since I don't think we care about the type in general - would we ever do something differently for an address than for an i32 or i64 (as appropriate) probably not. That means size in bytes is probably the right thing. The opcode choice in the function should be done with a switch where the default is an ASSERT_FATAL to make sure we catch oopses quickly.